src/Npgsql/TypeMapping/INpgsqlTypeMapper.cs

```diff
@@ -76,7 +88,10 @@ public interface INpgsqlTypeMapper
         /// Defaults to <see cref="NpgsqlSnakeCaseNameTranslator"/>
         /// </param>
         /// <typeparam name="TEnum">The .NET enum type to be mapped</typeparam>
-        INpgsqlTypeMapper MapEnum<TEnum>(string pgName = null, INpgsqlNameTranslator nameTranslator = null)
+        [NotNull]
+        INpgsqlTypeMapper MapEnum<TEnum>(
+            [CanBeNull] string pgName = null,
+            [CanBeNull] INpgsqlNameTranslator nameTranslator = null)
```

Comment:
ReSharper doesn't understand nullability by default values?

Reply:
Unless I'm mistaken, including these annotations here flows them onto implementation classes, which makes it simpler to ensure that concrete implementations all share the same semantics. 

It should be obvious to anyone looking at the interface that these can be null, but since it's a public interface, I think it makes sense to add them.

As an example, imagine that an implementation has a different default value for `pgName`, like `pgName = "test_name"`. With these annotations, ReSharper should still flow nullability info to the caller. But without these annotations, the nullability defaults to unknown (at least historically).

Reply:
For god's sake I hope C# 8 comes out soon with nullable references...